packages/hoppscotch-app/components/lenses/renderers/composables/useDownloadResponse.ts

```diff
@@ -0,0 +1,35 @@
+import { Ref, ref } from "@nuxtjs/composition-api"
+
+export type downloadResponseReturnType = (() => void) | Ref<any>
+
+export default function useDownloadResponse(
+  contentType: string,
+  responseBodyText: Ref<any>,
+  toast: any,
+  t: any
```

Comment:
Toast and i18n composables can be brought in using `useToast()` and `useI18n()` composables within the composable itself instead of passing it on.